stdlib/public/core/Array.swift

```diff
@@ -1865,3 +1869,189 @@ internal struct _ArrayAnyHashableBox<Element: Hashable>
     return true
   }
 }
+
+// SWIFT_ENABLE_TENSORFLOW
+extension Array where Element : Differentiable {
+  /// Views the array as the differentiable product manifold of `Element` multiplied with itself
+  /// `count` times.
+  @_fixed_layout
+  public struct DifferentiableView : Differentiable {
+    private var base: [Element]
+
+    /// The viewed array.
+    // I'm implementing this as a computed property instead of directly exposing `base` because the
+    // `@differentiable` annotation does not make the stored property actually differentiable. I
+    // think this is a bug. Maybe it's related to `@_fixed_layout`?
+    // TODO: Determine if that is a bug, and fix.
+    public var array: [Element] {
+      @differentiable(wrt: self, vjp: _vjpArray)
+      get {
```

Comment:
Use `_read` and `_modify` accessors instead of `get`/`set` to make it more efficient.

```suggestion
      _read { yield base }
      _modify { yield &base }
```